File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string) = logLine.[logLine.IndexOf(':') + 2..].Trim()

let logLevel (logLine: string) = logLine.[1..logLine.IndexOf(']') - 1].ToLower()

let reformat (logLine: string) = $"{message logLine} ({logLevel logLine})"

File: csharp/log-levels/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine) =>
        logLine[(logLine.IndexOf(':') + 1)..].Trim();

    public static string LogLevel(string logLine) =>
        logLine[1..logLine.IndexOf(']')].ToLower();

    public static string Reformat(string logLine) =>
        $"{Message(logLine)} ({LogLevel(logLine)})";
}
